src/CoreWCF.Templates/src/templates/CoreWCFService/Program.Main.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CoreWCFService
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseKestrel(options =>
+                    {
+                        options.AllowSynchronousIO = true;
+                    });
```

Comment:
This isn't needed any more unless you are using streamed transfer mode.

Reply:
it dropped it in [32d1a91](https://github.com/CoreWCF/CoreWCF/pull/590/commits/32d1a917a5c8ef67a8b585e09aa7a6b0febd0058)

Reply:
@FreeAndNil, it depends on what your contract looks like. If you are using data types which will use a serializer, then you will need to turn on `AllowSynchronousIO`. This is because we create an XmlWriter which writes directly to the network and pass that to the serializer and tell it to serialize. The serializers only have a synchronous implementation so write to the XmlWriter using synchronous api's only, which means the output stream only gets written to synchronously. If you are using `Stream` in your contract, I believe this doesn't need this option, but I haven't tested it. If it does still need it, I might be able to improve that.

Reply:
We use a stream as response.
We get no error, but after some time  (12 to 30 hours in our development environment) the server does no longer respond to new requests. We see many, many threads when looking at the dump.

I would propose opening a new issue.

Reply:
Please do open a new issue. Are you able to provide a process dump? You can contact me privately on gitter to arrange making a dump available.